### PR TITLE
Add feedback Implementation

### LINKED ISSUE #331: FEATURE: Feedback Implementation
### Is there an existing issue raised for this?

- [X] I have searched the existing issues

### Describe the feature

users submitting the feedback should get stored in the database associated with them. Optional implementation if needed i.e. sending thank you emails to the users.

Screenshot 2024-10-24 135807

### Component

backend/controllers/feedbackContoller

### It should be implemented because

Good feature fir a website like this and helps in testing the website for future upgrades and cutomizations

### Would you like to work on this issue?

None

### Terms & Conditions

- [X] I agree to follow the Code of Conduct
- [X] I'm a GSSOC'24 Ext contributor

### PR DESCRIPTION
**Pull request by haseebzaki-07**
## Fixes #331 

Removed the email implementation and made the feedback to reach the database.
Screenshot 2024-10-25 162341
Screenshot 2024-10-25 164041
Screenshot 2024-10-25 164129
Screenshot 2024-10-25 164201

## How should this be tested?

-Go to the feedback page, fill out the form and see the message.

### Required

- [X] Filled out the "How to test" section in this PR
- [X] Read Contributing guide
- [X] Self-reviewed my own code
- [X] Checked for warnings, there are none
- [X] Removed all `console.logs`
- [X] The changes don't cause any responsiveness issues

### CODE REVIEW

**Comment by haseebzaki-07** [comment]
hey @AlfiyaSiddique please review this

File: backend/server.js [hunk 1]
```diff
@@ -19,6 +19,7 @@ app.use(express.urlencoded({ limit: "10mb", extended: true }));
 const allowedOrigins = [
   "https://delightful-daifuku-a9f6ea.netlify.app",
   /https:\/\/deploy-preview-\d+--delightful-daifuku-a9f6ea\.netlify\.app/,
+  "http://localhost:5173"
 ];
```
Comment (AlfiyaSiddique) [inline backend/server.js:23]: @haseebzaki-07  remove this localhost URL

**Comment by AlfiyaSiddique** [comment]
@haseebzaki-07 one comment and you have deleted the .env.example file of backend folder please move that file back no need to delete just remove the envs that are not needed

**Comment by haseebzaki-07** [comment]
hey @AlfiyaSiddique deleted it by mistake, made the changes, please review

File: frontend/.env [hunk 2]
```diff
@@ -1,4 +1,4 @@
 # Not suggested modifying the backend port
 VITE_BACKEND_URL="http://localhost:8080"
 # Add your google client ID here
-VITE_GOOGLE_CLIENT_ID=""
+VITE_GOOGLE_CLIENT_ID="797632073512-kb63aipc56t835seibngtj9n5c76ar36.apps.googleusercontent.com"
```
Comment (AlfiyaSiddique) [inline frontend/.env:4]: @haseebzaki-07 remove your google client id

File: frontend/src/Pages/OneRecipe.jsx [hunk 3]
```diff
@@ -70,12 +70,12 @@ const OneRecipe = () => {
...
-    const username = JSON.parse(localStorage.getItem("username"));
+    const token = JSON.parse(localStorage.getItem("tastytoken"));
     if (commentInput.trim()) {
       try {
         await axios.post(`${backendURL}/api/recipe/addcomment`, {
           recipeId: recipe._id,
-          username: username,  // Assuming the user is already logged in
+          token: token,  // Assuming the user is already logged in
           content: commentInput,
         },
```
Comment (AlfiyaSiddique) [inline frontend/src/Pages/OneRecipe.jsx:80]: @haseebzaki-07 Do not assume, check if the user is logged in, If the user is logged in then only let them submit the feedback form.

**Comment by AlfiyaSiddique** [comment]
@haseebzaki-07 some more comments please fix

File: frontend/src/Pages/OneRecipe.jsx (hunk 3 above)
Comment (haseebzaki-07) [reply frontend/src/Pages/OneRecipe.jsx:80]: @AlfiyaSiddique the token is retrieved from the localStorage of the application, that is set only after the user logs in, so retreiveing the token means the user is logged in and is good to comment.

**Comment by haseebzaki-07** [comment]
hey @AlfiyaSiddique made the changes and please see me reply on the latest comment.

File: frontend/src/Pages/OneRecipe.jsx (hunk 3 above)
Comment (AlfiyaSiddique) [reply frontend/src/Pages/OneRecipe.jsx:80]: oaky great!

**Review by AlfiyaSiddique — APPROVED** [review]

**Comment by AlfiyaSiddique** [comment]
@haseebzaki-07 the deployment is failing

**Comment by AlfiyaSiddique** [comment]
image

**Comment by AlfiyaSiddique** [comment]
Please fix this and open a new PR will add the label to that PR

**Pull request merged**